File: benchmark-constraint-languages/layout.js
function fun() {
    var solver = new ClSimplexSolver();
    var obj = {gap: 0,
               pw: 0,
               lw: 0,
               rw: 0};
    bbb.always({solver: solver,  ctx: {obj: obj}}, function () { return obj.pw == 40000 });
    bbb.always({solver: solver,  ctx: {obj: obj}}, function () { return obj.gap == obj.pw / 20000 });
    bbb.always({solver: solver,  ctx: {obj: obj}}, function () { return obj.lw + obj.gap + obj.rw == obj.pw });
    bbb.always({solver: solver,  ctx: {obj: obj}}, function () { return obj.lw >= 0 });
    bbb.always({solver: solver,  ctx: {obj: obj}}, function () { return obj.rw >= 0 });
    console.log("gap " + obj.gap + ", left column " + obj.lw + ", right column " + obj.rw + ", page width " + obj.pw)
}

var start = Date.now();
for(var i = 0; i < 10; i++) {
    fun()
}
console.log("THIS IS THE TIME:" + (Date.now() - start));
alert.original.apply(window, ["CLOSE ME"])

File: benchmark-constraint-languages/layout.rb
require "libz3"

def fun
  gap, pw, lw, rw = [0]*4

  c1 = always { pw == 40000 }
  c2 = always { gap == pw / 20000 }
  c3 = always { lw + gap + rw == pw }
  c4 = always { lw >= 0 }
  c5 = always { rw >= 0 }

  puts "gap #{gap}, left column #{lw}, right column #{rw}, page width #{pw}"
  [c1,c2,c3,c4,c5].each(&:disable)
end

start = Time.now
10.times do
  fun
end
puts "THIS IS THE TIME: #{Time.now - start}"
